tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/smt/Z3SolverContext.scala

```diff
@@ -366,31 +365,42 @@ class Z3SolverContext(val config: SolverConfig) extends SolverContext with LazyL
     }
   }
 
-  private def initLog(): PrintWriter =
-    OutputManager.runDirPathOpt match {
-      case None => new PrintWriter(NullOutputStream.NULL_OUTPUT_STREAM)
-      case Some(_) => {
-        val writer = OutputManager.printWriter(OutputManager.runDir, s"log$id.smt")
-        if (!config.debug) {
-          writer.println("Logging is disabled (Z3SolverContext.debug = false). Activate with --debug.")
-          writer.flush()
-        }
-        writer
+  /**
+   * Initialize up to two log writers, one in the run directory and one in the custom run directory, if those are set.
+   */
+  private def initLogs(): Seq[PrintWriter] = {
+    val filePart = s"log$id.smt"
+    val writers = Seq(OutputManager.runDirPathOpt, OutputManager.customRunDirPathOpt).flatten.map(
```

Comment:
You can `++` two options to get an iterable:
```suggestion
    val writers =(OutputManager.runDirPathOpt ++ OutputManager.customRunDirPathOpt).map(
```
looks nicer IMO, but it's more of a preference.